### PR TITLE
CLI Validation Tests

### PR DESCRIPTION
**Pull request by waymobetta**
## Description

1. This PR introduces unit tests for the `Validate[command_name]Flags` functions present within _most_ CLI commands.

2. It also introduces a few of these validation functions that previously did not exist.

3. It also renames a flag: `destination` --> `recipient`; found within the `ERC20` package.
a. this keeps flag naming consistent between the `ERC20` and `ERC721` packages.

Considerations:
- I actually wasn't sure whether this is the best way to do this. It seems fine, looks a bit redundant, but I figured that it would be better to have more than less as we can always delete what we think is unnecessary.
- I actually split up the tests into multiple files within the `account` package since there were only a few commands.
a. it seemed unnecessary to follow this pattern for the other packages (given the number of files/commands) which is why the testable functions are largely encapsulated within one file per package.

Open to feedback!
## Related Issue Or Context

Closes: #199 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [x] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -563,7 +563,7 @@ Flags:
       --amount string        Deposit amount
       --contract string      ERC20 contract address
       --decimals uint        ERC20 token decimals
-      --destination string   Where tokens should be minted. Defaults to TX sender
+      --recipient string     Recipient address
```
Comment (waymobetta) [inline README.md:566]: I actually prefer `--destination` but changed it to be consistent with the [ERC721.Mint](https://github.com/ChainSafe/chainbridge-core/blob/waymobetta/cli-validation-tests/chains/evm/cli/erc721/mint.go#L58). 

Open to discussion!

File: chains/evm/cli/account/import-priv-key.go [hunk 2]
```diff
@@ -25,6 +33,12 @@ func BindImportPrivKeyFlags(cmd *cobra.Command) {
 func init() {
 	BindImportPrivKeyFlags(importPrivKeyCmd)
 }
+func ValidateImportPrivKeyFlags(cmd *cobra.Command, args []string) error {
+	if len(PrivateKey) != 64 {
+		return fmt.Errorf("invalid private key: length")
+	}
+	return nil
+}
```
Comment (waymobetta) [inline chains/evm/cli/account/import-priv-key.go:41]: _Some_ validation for PK length already exists within the `secp256k1` package though perhaps it is better to present user with a message before it even gets that far

**Review by P1sar — APPROVED** [review]

**Comment by MakMuftic** [comment]
This looks great. As I was going thru code I remembered one interesting pattern that we used before on some old project and it would be very applicable to tests like these ones. The intention of this is not to refactor these tests, as there is nothing wrong with them, just wanted to share an interesting approach when you have repetitive testing logic inside unit tests - maybe we use it on some other unit tests. @P1sar @waymobetta @nmlinaric @tcar121293 @mpetrun5 

You can find a basic idea in the snippet below.
```go


func (s *ERC721TestSuite) TestErc721CLIFlagValidation() {
	// these types would be defined somewhere else
	type flags struct {
		name  string
		value string
	}
	type ValidationFunc func(cmd *cobra.Command, args []string) error
	type BindFunc func(cmd *cobra.Command)
	
	// test cases
	tests := []struct {
		flags              []flags
		bindFunction       BindFunc
		validationFunction ValidationFunc
		err                bool
	}{
		{
			flags: []flags{
				{name: "contract", value: validAddr},
				{name: "minter", value: validAddr},
			},
			bindFunction:       BindAddMinterFlags,
			validationFunction: ValidateAddMinterFlags,
			err:                false,
		},
		{
			flags: []flags{
				{name: "contract", value: invalidAddr},
				{name: "minter", value: invalidAddr},
			},
			bindFunction:       BindAddMinterFlags,
			validationFunction: ValidateAddMinterFlags,
			err:                true,
		},
		{
			flags: []flags{
				{name: "contract", value: validAddr},
				{name: "recipient", value: validAddr},
			},
			bindFunction:       BindApproveFlags,
			validationFunction: ValidateApproveFlags,
			err:                false,
		},
		{
			flags: []flags{
				{name: "contract", value: invalidAddr},
				{name: "recipient", value: invalidAddr},
			},
			bindFunction:       BindApproveFlags,
			validationFunction: ValidateApproveFlags,
			err:                true,
		},
	}
	
	// test logic
	for _, tc := range tests {
		cmd := new(cobra.Command)
		BindAddMinterFlags(cmd)

		for _, f := range tc.flags {
			err := cmd.Flag(f.name).Value.Set(f.value)
			s.Nil(err)
		}

		err = tc.validationFunction(
			cmd,
			[]string{},
		)
		if tc.err {
			s.NotNil(err)
		} else {
			s.Nil(err)
		}
	}
}

```

**Review by MakMuftic — APPROVED** [review]

**Comment by waymobetta** [comment]
@MakMuftic this is an awesome suggestion. I've actually seen this pattern before as well and was racking my brain to remember where--just found it!

[chainbridge-celo/account_test.go](https://github.com/ChainSafe/chainbridge-celo/blob/main/account_test.go#L139-L176)

**Pull request merged**